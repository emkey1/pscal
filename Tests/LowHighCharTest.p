program LowHighCharTest;

procedure AssertEqualChar(expected, actual: char; testName: string);
begin
  write('START: ', testName, ': ');
  if expected = actual then
    writeln('PASS')
  else
    writeln('FAIL (expected: ''', expected, ''', got: ''', actual, ''')');
end;

begin
  AssertEqualChar(chr(0), low(char), 'low(char)');
  AssertEqualChar(chr(255), high(char), 'high(char)');
end.
