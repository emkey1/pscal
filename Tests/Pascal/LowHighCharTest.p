program LowHighCharTest;

type
  TColor = (Red, Green, Blue);

procedure AssertEqualChar(expected, actual: char; testName: string);
begin
  write('START: ', testName, ': ');
  if expected = actual then
    writeln('PASS')
  else
    writeln('FAIL (expected: ''', expected, ''', got: ''', actual, ''')');
end;

procedure AssertEqualColor(expected, actual: TColor; testName: string);
begin
  write('START: ', testName, ': ');
  if ord(expected) = ord(actual) then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', ord(expected), ', got: ', ord(actual), ')');
end;

var
  cLow, cHigh: TColor;

begin
  AssertEqualChar(chr(0), low(char), 'low(char)');
  AssertEqualChar(chr(255), high(char), 'high(char)');
  cLow := low(TColor);
  cHigh := high(TColor);
  AssertEqualColor(Red, cLow, 'low(TColor)');
  AssertEqualColor(Blue, cHigh, 'high(TColor)');
end.
