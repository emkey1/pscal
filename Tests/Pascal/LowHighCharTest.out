START: low(char): PASS
START: high(char): PASS
START: low(TColor): PASS
START: high(TColor): PASS
